hack/generate-extension.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+# SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+set -o pipefail
+
+function usage {
+    cat <<EOM
+Usage:
+generate-extension [options]
+
+This script generates a Extension (operator.gardener.cloud) manifest. Kustomize and YQ are required to run this script.
+
+    -h, --help                              Display this help and exit.
+    --name                                  Name is the name of the extension.
+    --provider-type                         Type of the provider.
+    --component-name                        Name of the Kustomize component, one of {provider-extension, network, containerruntime, extension}
```

Comment:
Component name is somewhat misleading. I would prefer to name it e.g. component category.
```suggestion
    --component-category                        Category of the Kustomize component, one of {provider-extension, network, containerruntime, extension}
```